server/src/main/java/org/elasticsearch/script/StringFieldScript.java

```diff
@@ -81,22 +81,22 @@ public StringFieldScript(String fieldName, Map<String, Object> params, SearchLoo
         super(fieldName, params, searchLookup, ctx);
     }
 
-    /**
-     * Execute the script for the provided {@code docId}.
-     *
-     * @return a mutable {@link List} that contains the results of the script
-     * and will be modified the next time you call {@linkplain #resultsForDoc}.
-     */
-    public final List<String> resultsForDoc(int docId) {
+    @Override
+    protected void prepareExecute() {
         results.clear();
         chars = 0;
-        setDocument(docId);
-        execute();
-        return results;
     }
 
     public final void runForDoc(int docId, Consumer<String> consumer) {
```

Comment:
I'm curious about this variant of the `runForDoc(int docId)` method we are centralizing in this PR. So far I only see it in StringFieldScript and therefore its subtypes. Is there anything special about it that we need this extended "consumer" arg here?

Reply:
It's been bothering me a bit. We are not centralizing it because the consumer argument takes the type that the script emits. We have it in other places too. I may look further to see if we can do something about it and whether it is still useful.